From 29ef1977d1d62a66189e78b5055406d7c7a7b709 Mon Sep 17 00:00:00 2001
From: Thomas Arcila <thomas.arcila@gmail.com>
Date: Tue, 19 Jan 2021 11:25:14 +0100
Subject: [PATCH 1/2] strutil: Use external fmtlib

---
 src/include/OpenImageIO/strutil.h | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/include/OpenImageIO/strutil.h b/src/include/OpenImageIO/strutil.h
index 82299999..992e3c5d 100644
--- a/src/include/OpenImageIO/strutil.h
+++ b/src/include/OpenImageIO/strutil.h
@@ -41,9 +41,9 @@
 #ifndef FMT_USE_GRISU
 #    define FMT_USE_GRISU 1
 #endif
-#include "detail/fmt/ostream.h"
-#include "detail/fmt/format.h"
-#include "detail/fmt/printf.h"
+#include "fmt/ostream.h"
+#include "fmt/format.h"
+#include "fmt/printf.h"
 #if OIIO_GNUC_VERSION >= 70000
 #    pragma GCC diagnostic pop
 #endif
-- 
2.32.0

