From 82d366b0dd809c788d530e893530eea6aa2083b6 Mon Sep 17 00:00:00 2001
From: Thomas Arcila <thomas.arcila@gmail.com>
Date: Sun, 21 Aug 2022 17:31:38 -0400
Subject: [PATCH] View: Adapt search path so it works with Bazel's runtime
 files

---
 source/MaterialXView/Main.cpp | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/source/MaterialXView/Main.cpp b/source/MaterialXView/Main.cpp
index 18d5122..644e0c3 100644
--- a/source/MaterialXView/Main.cpp
+++ b/source/MaterialXView/Main.cpp
@@ -63,9 +63,18 @@ mx::FileSearchPath getDefaultSearchPath()
     mx::FilePath modulePath = mx::FilePath::getModulePath();
     mx::FilePath installRootPath = modulePath.getParentPath();
     mx::FilePath devRootPath = installRootPath.getParentPath().getParentPath();
+    // Bazel case, where the binary is run from the folder holding the runfiles.
+    mx::FilePath bazelSearchPath = mx::FilePath::getCurrentPath();
+    mx::FilePath bazelSearchPathExternal = mx::FilePath::getCurrentPath() / "external" / "materialx";
 
     mx::FileSearchPath searchPath;
-    if ((devRootPath / "libraries").exists())
+    if ( (bazelSearchPath / "libraries").exists()) {
+        searchPath.append(bazelSearchPath);
+    }
+    else if ( (bazelSearchPathExternal / "libraries").exists()) {
+        searchPath.append(bazelSearchPathExternal);
+    }
+    else if ((devRootPath / "libraries").exists())
     {
         searchPath.append(devRootPath);
     }
-- 
2.37.2

