diff --git a/build/version_info_windows.js b/build/version_info_windows.js
index d3f4c0f..e774f73 100644
--- a/build/version_info_windows.js
+++ b/build/version_info_windows.js
@@ -49,11 +49,10 @@ if ( WScript.Arguments(0).toLowerCase().match("gcc") ) {
 } else { // MS / Intel compilers
     //Getting BUILD_CL
     tmpExec = WshShell.Exec("cmd /c echo #define 0 0>empty.cpp");
-    tmpExec = WshShell.Exec("cl -c empty.cpp ");
     while ( tmpExec.Status == 0 ) {
         WScript.Sleep(100);
     }
-    var clVersion = tmpExec.StdErr.ReadLine();
+    var clVersion = "cl";
     WScript.echo( "#N \": BUILD_CL\\t\\t" + 
                   clVersion +
                   "\" ENDL \\" );
