From b9199abd7384bc484992c4d5e5fb5131feef6dbd Mon Sep 17 00:00:00 2001
From: Frederic Devernay <deverf@amazon.com>
Date: Tue, 1 Dec 2020 18:44:43 -0800
Subject: [PATCH 4/4] matrix decompose(): fix bug

---
 glm/gtx/matrix_decompose.inl | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/glm/gtx/matrix_decompose.inl b/glm/gtx/matrix_decompose.inl
index c66a9987..8aec64ba 100644
--- a/glm/gtx/matrix_decompose.inl
+++ b/glm/gtx/matrix_decompose.inl
@@ -174,10 +174,10 @@ namespace detail
 
 			root = sqrt(Row[i][i] - Row[j][j] - Row[k][k] + static_cast<T>(1.0));
 
-			Orientation[i] = static_cast<T>(0.5) * root;
+			Orientation[i + 1] = static_cast<T>(0.5) * root;
 			root = static_cast<T>(0.5) / root;
-			Orientation[j] = root * (Row[i][j] + Row[j][i]);
-			Orientation[k] = root * (Row[i][k] + Row[k][i]);
+			Orientation[j + 1] = root * (Row[i][j] + Row[j][i]);
+			Orientation[k + 1] = root * (Row[i][k] + Row[k][i]);
 			Orientation.w = root * (Row[j][k] - Row[k][j]);
 		} // End if <= 0
 
-- 
2.30.1.windows.1

